addons/microsoft_calendar/models/calendar.py

```diff
@@ -82,11 +82,12 @@ def create(self, vals_list):
         if self._check_microsoft_sync_status() and not notify_context and recurrency_in_batch:
             self._forbid_recurrence_creation()
 
-        for vals in vals_list:
-            # If event has a different organizer, check its sync status and verify if the user is listed as attendee.
-            sender_user, partner_ids = self._get_organizer_user_change_info(vals)
-            partner_included = partner_ids and len(partner_ids) > 0 and sender_user.partner_id.id in partner_ids
-            self._check_organizer_validation(sender_user, partner_included)
+        if self._check_organizer_validation_conditions(vals_list.get("appointment_type_id", None)):
+            for vals in vals_list:
+                # If event has a different organizer, check its sync status and verify if the user is listed as attendee.
+                sender_user, partner_ids = self._get_organizer_user_change_info(vals)
+                partner_included = partner_ids and len(partner_ids) > 0 and sender_user.partner_id.id in partner_ids
+                self._check_organizer_validation(sender_user, partner_included)
```

Comment:
As I remember this is because microsoft requires it, is it really ok to skip it merely because it's an appointment on our side?

If so great but it feels like it might be necessary to have the organizer as attendee in appointment,.

Reply:
Hey @reth-odoo I don't know that it matters because from my testing on older code without this check in place, the calendar would just not sync that event if the organizer was not an attendee. Which should be fine in this case. I have tested this on old code and this is how the functionality worked before this commit https://github.com/odoo/odoo/commit/a2e823858bbd6d5199d71ad40747d6e633dd98fc

I am not sure if this is the proper functionality, but it seems better than giving customers unwanted emails when their customers book a resource.

Reply:
Indeed, seems ok if it's handles it fine :)

Reply:
Note that in https://github.com/odoo/odoo/pull/129017 we expect that when syncing, all events will be synced and send an email to the participants. So if it's not the case here we'll need to adapt the logic a bit.

Maybe just updating `_check_microsoft_sync_status` to specify we won't sync for this case

Reply:
Thanks for the heads up @reth-odoo. Haven't had a chance to work on this for a bit, but I am making some changes now.